lib/cancan/model_adapters/active_record_5_adapter.rb

```diff
@@ -21,17 +21,57 @@ def self.matches_condition?(subject, name, value)
 
       private
 
+      delegate :connection, :quoted_primary_key, to: :@model_class
+      delegate :quote_table_name, to: :connection
+
       def build_joins_relation(relation, *where_conditions)
         case CanCan.accessible_by_strategy
         when :subquery
-          inner = @model_class.unscoped do
-            @model_class.left_joins(joins).where(*where_conditions)
-          end
-          @model_class.where(@model_class.primary_key => inner)
-
+          build_joins_relation_subquery(where_conditions)
```

Comment:
This refactor was needed to deal with Rubocop complexity offences.

Reply:
we should probably, at some point, extract these strategies

Reply:
@coorasse I have extracted all strategies as separate classes (also to solve Rubocop complexity issues). What do you think?